src/java/org/apache/cassandra/index/sai/memory/TrieMemoryIndex.java

```diff
@@ -111,6 +114,39 @@ public TrieMemoryIndex(IndexContext indexContext, Memtable memtable, AbstractBou
         this.data = InMemoryTrie.longLived(TypeUtil.BYTE_COMPARABLE_VERSION, TrieMemtable.BUFFER_TYPE, indexContext.columnFamilyStore().readOrdering());
         this.primaryKeysReducer = new PrimaryKeysReducer();
         this.memtable = memtable;
+        termFrequencies = new ConcurrentHashMap<>();
+    }
+
+    public synchronized Map<PrimaryKey, Integer> getDocLengths()
+    {
+        return docLengths;
+    }
+
+    private static class PkWithTerm
+    {
+        private final PrimaryKey pk;
+        private final ByteComparable term;
+
+        private PkWithTerm(PrimaryKey pk, ByteComparable term)
+        {
+            this.pk = pk;
+            this.term = term;
+        }
+
+        @Override
+        public int hashCode()
+        {
+            return Objects.hash(pk, ByteComparable.length(term, ByteComparable.Version.OSS41));
```

Comment:
Nit: I think we should use `TypeUtil.BYTE_COMPARABLE_VERSION` here and below. We configure the trie to use that as the version, and that will ensure the two are always in sync.